pkg/webhook/pod/pd_deleter.go

```diff
@@ -191,12 +193,22 @@ func (pc *PodAdmissionControl) transferPDLeader(payload *admitPayload) *admissio
 		return util.ARFail(err)
 	}
 	tcName := payload.tc.Name
-	lastOrdinal := payload.tc.Status.PD.StatefulSet.Replicas - 1
 	var targetName string
-	if ordinal == lastOrdinal {
-		targetName = pdutil.PdPodName(tcName, 0)
+
+	if features.DefaultFeatureGate.Enabled(features.AdvancedStatefulSet) {
+		lastOrdinal := helper.GetMaxPodOrdinal(*payload.ownerStatefulSet.Spec.Replicas, payload.ownerStatefulSet)
+		if ordinal == lastOrdinal {
+			targetName = pdutil.PdPodName(tcName, helper.GetMinPodOrdinal(*payload.ownerStatefulSet.Spec.Replicas, payload.ownerStatefulSet))
+		} else {
+			targetName = pdutil.PdPodName(tcName, lastOrdinal)
+		}
```

Comment:
actually, this works both with Kubernetes Statefulset and Advanced Statefulset because ast helpers will return correct results for built statefulset too. or you prefer using ats helpers only when the feature is enabled?

Reply:
Good catch, I will merge both cases to make it more simple.